<launch>

	<arg name="camera" default="kinect1"/>
	<arg name="online" default="true"/>
	<arg name="bus" default="0"/>
	<arg name="filename"/>

	<group if="$(arg online)">
		<param name="use_sim_time" value="false" />
		<include file="$(find openni_launch)/launch/openni.launch">
			<arg name="camera" value="$(arg camera)" />
<!--			<arg name="device_id" value="$(arg bus)@0" /> -->
			<arg name="depth_registration" value="true" />
		</include>
	</group>
	<group unless="$(arg online)">
		<param name="use_sim_time" value="true" />
		<include file="play_cloud.launch">
			<arg name="camera" value="$(arg camera)" />
			<arg name="filename" value="$(arg filename)" />
		</include>
	</group>

	<node name="drop1" pkg="topic_tools" type="drop" args="/$(arg camera)/depth_registered/points 2 3 /drop/points"/>
<group ns="preprocessor">>
 <param name="min_hue" value="0"/>
 <param name="max_hue" value="255"/>
 <param name="min_hue" value="0"/>
 <param name="max_hue" value="255"/>
 <param name="min_hue" value="0"/>
 <param name="max_hue" value="255"/>
</group>
	<node name="preprocessor" pkg="binary_symlinks" type="simple_preprocessor_node" output="screen"  args="--inputTopic=/drop/points --zClipLow=.4 --downsample=.01 --clusterMinSize=0"/>
</launch>
